Toggle local verification with -localVerify

yondonfu opened the pull request:
**What does this pull request do? Explain your changes. (required)**
<!-- A clear and concise description of what this pull request does. -->

This PR adds a `-localVerify` flag with the following behavior:

- If the node is in on-chain mode, the default is `localVerify = true`
- If the node is in off-chain mode, the default is `localVerify = false`
- A user can also explicitly set `-localVerify=true|false`
- When `localVerify = true`, the node will enable pixel count and signature verification
- When `-verifierURL != ""`, the node will also enable pixel count and signature verification
- When `localVerify = false`, the node will disable pixel count and signature verification

This flag should be useful when the node is running in on-chain mode, but working with trusted orchestrators. In this case, the node does not need to incur additional overhead for local verification.

This PR also adds docs on verification that provides a high level overview on how to enable/disable local + tamper verification.

**Specific updates (required)**
<!--- List out all significant updates your code introduces -->

See commit history.

**How did you test each of these updates (required)**
<!-- A detailed description of how you tested your code changes. Include details of your testing environment, and the tests you ran to see how your change affects other areas of the code, etc. -->

Updated `test_args.sh` and manually tested.

**Checklist:**
<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
- [x] README and other documentation updated
- [x] Node runs in OSX and devenv
- [x] All tests in `./test.sh` pass

yondonfu committed:
cmd: Toggle local verification with -localVerify

yondonfu committed:
doc: Add verification docs

j0sh reviewed (approved):
LGTM! 🚢

j0sh commented:
Might be nice to add some checks to `test_args.sh` but up to you.

kyriediculous commented:
Sorry ignore my previous comment.

Why does this need a flag again? I feel like this should be enabled regardless. 

Tamper verification can be an optional addition

darkdarkdragon reviewed (approved):
LGTM

yondonfu committed:
cmd: Add log when local verification is enabled

Include tests in test_args.sh based on the local verification log
statement on node startup.

yondonfu commented:
> Might be nice to add some checks to test_args.sh but up to you.

Added tests in `test_args.sh` in 37c1af8

> Why does this need a flag again? I feel like this should be enabled regardless.

@kyriediculous The flag allows a broadcaster operator to disable local verification i.e. pixel count (used for checking if the broadcaster is charged properly based on # of pixels processed) + signature verification (used for gathering signed evidence which can be submitted to a third party if a tamper is detected) when it is interacting with orchestrators that it trusts. 

This PR already has an approval, but I'll let it sit for another 2 hours and then I'll merge it.
